### PR TITLE
Update Scala to version 3

### PR DESCRIPTION
**Pull request by jpsacha**
Scala updated to latest release 3.2.2. Added some additional tests to verify operation.

### CODE REVIEW

File: src/main/scala/org/scijava/plugins/scripting/scala/ScalaAdaptedScriptEngine.scala [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  override def get(key: String): AnyRef =
+    // First try to get value from bindings
+    var value = super.get(key)
+
+    // NB: Extracting values from Scala Script Engine are a little tricky.// NB: Extracting values from Scala Script Engine are a little tricky.
+    // Values (variables) initialised or computed in the script are// Values (variables) initialised or computed in the script are
+    // not added to the bindings of the CompiledScript AFAICT. Therefore// not added to the bindings of the CompiledScript AFAICT. Therefore
+    // the only way to extract them is to evaluate the variable and// the only way to extract them is to evaluate the variable and
+    // capture the return. If it evaluates to null or throws a// capture the return. If it evaluates to null or throws a
+    // a ScriptException, we simply return null.// a ScriptException, we simply return null.
```
Comment (imagejan) [inline src/main/scala/org/scijava/plugins/scripting/scala/ScalaAdaptedScriptEngine.scala]: The comments here are duplicated.

Comment (jpsacha) [reply src/main/scala/org/scijava/plugins/scripting/scala/ScalaAdaptedScriptEngine.scala]: Corrected

**Comment by jpsacha** [comment]
One more fix for a warning in Script Editor when empty value is returned.

**Comment by ctrueden** [comment]
Thank you @jpsacha for updating this, and sorry @keithschulze that #6 fell through the cracks.

I have released scripting-scala 0.3.0 containing this update.

**Pull request merged**